kimchi/src/circuits/polynomials/poseidon.rs

```diff
@@ -10,28 +10,34 @@ use std::marker::PhantomData;
 use CurrOrNext::*;
 
 /// An equation of the form `(curr | next)[i] = round(curr[j])`
-pub struct RoundEquation {
+struct RoundEquation {
     pub source: usize,
     pub target: (CurrOrNext, usize),
 }
 
-pub const ROUND_EQUATIONS: [RoundEquation; ROUNDS_PER_ROW] = [
+//~ A poseidon gate records 5 rounds of the poseidon permutation:
+const ROUND_EQUATIONS: [RoundEquation; ROUNDS_PER_ROW] = [
+    //~ * the first round targets the first row
```

Comment:
This comment could easily say `The first four rounds target the first row`, rather than copy-pasting below. It's probably also relevant to mention the layout beyond just the row.

Reply:
Actually I was just toying around on how to specify poseidon, I think it’s going to be hard due to how it’s implemented, but ignore most of that for now